flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/DataSetPreAggFunction.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.runtime.aggregate
+
+import java.lang.Iterable
+
+import org.apache.flink.api.common.functions._
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.table.functions.{Accumulator, AggregateFunction}
+import org.apache.flink.types.Row
+import org.apache.flink.util.{Collector, Preconditions}
+
+/**
+  * [[GroupCombineFunction]] and [[MapPartitionFunction]] to compute pre-aggregates for batch
+  * (DataSet) queries.
+  *
+  * @param aggregates The aggregate functions.
+  * @param aggInFields The positions of the aggregation input fields.
+  * @param groupingKeys The positions of the grouping keys in the input.
+  */
+class DataSetPreAggFunction(
+    private val aggregates: Array[AggregateFunction[_ <: Any]],
+    private val aggInFields: Array[Int],
+    private val groupingKeys: Array[Int])
+  extends AbstractRichFunction
+    with GroupCombineFunction[Row, Row]
```

Comment:
Can we align with and extend? 
The suggestion just ref to: [Scalar Class/Object/Trait constructors declarations ](http://docs.scala-lang.org/style/declarations)